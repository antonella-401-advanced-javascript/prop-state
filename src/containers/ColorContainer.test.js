import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, mount } from 'enzyme';
import ColorContainer from './ColorContainer';

Enzyme.configure({ adapter: new Adapter() });
describe('Color Container', () => {
  it('renders Color Container', () => {
    const wrapper = shallow(<ColorContainer />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Changes Color on button click', () => {
    const wrapper = mount(<ColorContainer colors={[
      'red',
      'green',
      'black',
      'yellow',
    ]} />);
    wrapper.findWhere(el => el.key() === 'red').simulate('click');
    expect(wrapper.state('backgroundColor')).toBe('red');
  });
});