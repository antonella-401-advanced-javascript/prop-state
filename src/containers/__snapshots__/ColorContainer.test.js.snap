// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color Container renders Color Container 1`] = `
<Fragment>
  <ColorDisplay
    backgroundColor="black"
  />
  <ColorPicker
    colors={
      Array [
        "red",
        "green",
        "black",
        "yellow",
      ]
    }
    selectColor={[Function]}
  />
</Fragment>
`;
